### PR TITLE
Add blog 2024-01-17

### PR DESCRIPTION
**Pull request by MythicalFish**

### CODE REVIEW

File: content/blog/2024/optimizing-nextflow-for-hpc-and-cloud-at-scale.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+icon=ben.jpg
+~~~~~~
+
+## Introduction
+
+A Nextflow workflow run consists of the head job (Nextflow itself) and compute tasks (defined in the pipeline script). It is common to request resources for the tasks via process directives such as `cpus` and `memory`, but the Nextflow head job also requires compute resources. Most of the time, users don’t need to explicitly define the head job resources, as Nextflow generally does a good job of allocating resources for itself. For very large workloads, however, head job resource sizing becomes much more important.
+
+In this article, we will help you understand how the Nextflow head job works and show you how to tune head job resources such as CPUs and memory for your use case.
+
+## Head job resources
```
Comment (ewels) [inline content/blog/2024/optimizing-nextflow-for-hpc-and-cloud-at-scale.md:18]: Good place to truncate the intro blurb on the listing page here..

```suggestion
In this article, we will help you understand how the Nextflow head job works and show you how to tune head job resources such as CPUs and memory for your use case.

<!-- end-archive-description -->

## Head job resources
```

**Review by ewels — APPROVED** [review]
Nice! Couple of minor suggestions but nothing big.

File: content/blog/2024/optimizing-nextflow-for-hpc-and-cloud-at-scale.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+### CPUs
+
+Nextflow uses a thread pool to run native Groovy code (e.g. channel operators, `exec` processes), submit tasks to executors, and publish output files. The number of threads is based on the number of available CPUs, so if you want to provide more compute power to the head job, simply allocate more CPUs and Nextflow will use them. In Nextflow Tower, you can use **Head Job CPUs** or **Head Job submit options** (depending on the compute environment) to allocate more CPUs.
```
Comment (bentsherman) [inline content/blog/2024/optimizing-nextflow-for-hpc-and-cloud-at-scale.md:24]:
```suggestion
Nextflow uses a thread pool to run native Groovy code (e.g. channel operators, `exec` processes), submit tasks to executors, and publish output files. The number of threads is based on the number of available CPUs, so if you want to provide more compute power to the head job, simply allocate more CPUs and Nextflow will use them. In the Seqera Platform, you can use **Head Job CPUs** or **Head Job submit options** (depending on the compute environment) to allocate more CPUs.
```

File: content/blog/2024/optimizing-nextflow-for-hpc-and-cloud-at-scale.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+That being said, this benchmark consisted of only two runs of nf-core/rnaseq. We didn’t perform more runs here because they were so large, so your results may vary. In particular, if your Nextflow runs spend a lot of time publishing outputs after all the compute tasks have completed, you will likely benefit the most from using virtual threads. In any case, virtual threads should perform at least as well as platform threads, albeit with higher memory usage in some cases.
+
+## Summary
+
+The key to right-sizing the Nextflow head job is to understand which parts of a Nextflow pipeline are executed directly by Nextflow, and which parts are delegated to compute tasks. This knowledge will help prevent head job failures at scale.
+
+Here are the main takeaways:
+
+- Nextflow uses a thread pool based on the number of available CPUs.
+- Nextflow uses a maximum heap size based on the standard JVM options, which is typically 25% of physical memory (75% in Tower).
```
Comment (bentsherman) [inline content/blog/2024/optimizing-nextflow-for-hpc-and-cloud-at-scale.md:197]:
```suggestion
- Nextflow uses a maximum heap size based on the standard JVM options, which is typically 25% of physical memory (75% in the Seqera Platform).
```

File: assets/img/blog-2024-01-17--performance-graphs.jpg
[the remark this answers is not in the document]
Comment (MythicalFish) [reply assets/img/blog-2024-01-17--performance-graphs.jpg:1]: Hmm as a PNG it's 4x the size 🤔
If you happen to know where I can source an SVG version I could update it later on

[the remark this answers is not in the document]
Comment (bentsherman) [reply assets/img/blog-2024-01-17--performance-graphs.jpg:1]: Looks like he took a screenshot of the google doc, I will add the original PNGs

**Comment by bentsherman** [comment]
@MythicalFish I will upload the original PNGs and then we should be good to go

**Comment by MythicalFish** [comment]
Thanks @bentsherman !

**Pull request merged**